tklock.c

```diff
@@ -294,6 +304,81 @@ static void disable_autorelock_policy(void)
 	return;
 }
 
+/**
+ * Enable/disable X11 input events (currently only for touchscreen)
+ *
+ * @param enable TRUE enable events, FALSE disable events
+ * @return TRUE on success, FALSE on failure
+ */
+static gboolean generic_x11_event_control(const gboolean enable)
+{
+	gboolean ret = FALSE;
+	Display* dpy = NULL;
+	XDeviceInfo *devinfo = NULL;
+	int ndev = 0;
+	unsigned char data = (unsigned char)enable;
+
+	dpy = XOpenDisplay(NULL);
+	if (dpy == NULL) {
+		dpy = XOpenDisplay(":0.0");
+	}
+
+	if (dpy == NULL) {
+		mce_log(LL_INFO, "%s: unable to open display", __func__);
+		return ret;
+	}
```

Comment:
I'd replace with:
```
	if (dpy == NULL) {
		if ((dpy = XOpenDisplay(":0.0")) == NULL) {
			mce_log(LL_INFO, "%s: unable to open display", __func__);
			return ret;
		}
	}
```

Reply:
It allows to get rid of 2 same if's in a row:
```
if (dpy == NULL) ...
if (dpy == NULL) ...
```

Reply:
In some earlier PR you were asking me to do the opposite: not have assignments in if statements, because that was less clear.

The display opening is currently done in the same manner as in the modules/display PR I made earlier, so I am not sure if I want to rewrite that now.

Reply:
In the earlier PR there was a 3 lines big statement. That's why I asked you to not overcomplicate it even more. In this case it's fine.

If you don't like the suggestion, here's the way to go:
```
	if (dpy == NULL) {
		dpy = XOpenDisplay(":0.0");

		if ((dpy == NULL) {
			mce_log(LL_INFO, "%s: unable to open display", __func__);
			return ret;
		}
	}
```

The point is not to have 2 equal if's in a row (see above comment).

Reply:
> The display opening is currently done in the same manner as in the modules/display PR I made earlier, so I am not sure if I want to rewrite that now.

Yep, missed it there, sorry.
Could you please add the fix for that case as well?
Or I can do it in my branch.

Reply:
Maybe we can just keep it as is in both places? I think it's quite clear with the extra if statement.

Reply:
Agree, it is more clear with two independent if()s

Reply:
ok